statsmodels/tsa/stattools.py

```diff
@@ -467,7 +467,10 @@ def q_stat(x, nobs, type="ljungbox"):
     if type == "ljungbox":
         ret = (nobs * (nobs + 2) *
                np.cumsum((1. / (nobs - np.arange(1, len(x) + 1))) * x**2))
-    chi2 = stats.chi2.sf(ret, np.arange(1, len(x) + 1))
+
+    with np.errstate(invalid='ignore'):
+        # Suppress warnings caused by NaNs in `ret`
+        chi2 = stats.chi2.sf(ret, np.arange(1, len(x) + 1))
```

Comment:
why is there a nan?
is the nan intentional or a bug?

Reply:
(I think I intended to do changes outside of tests/ files in separate PR(s), not sure how this snuck in)

This happens if the input `x` includes `NaN`s.  Since `x` is often produced with a call to `acf`, this can definitely happen in non-bug ways.

The warnings this produces when hit do not make their source easy to track down:

```
>>> chi2 = stats.chi2.sf(ret, np.arange(1, len(x) + 1))
/usr/local/lib/python2.7/site-packages/scipy/stats/_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in greater
  return (self.a < x) & (x < self.b)
/usr/local/lib/python2.7/site-packages/scipy/stats/_distn_infrastructure.py:877: RuntimeWarning: invalid value encountered in less
  return (self.a < x) & (x < self.b)
/usr/local/lib/python2.7/site-packages/scipy/stats/_distn_infrastructure.py:1831: RuntimeWarning: invalid value encountered in less_equal
  cond2 = cond0 & (x <= self.a)
```

If we do want a warning to be issued, which is not unreasonable, we should issue it directly from this function at the input-validation step.

Reply:
Easy way to track is to turn the warning (temporarily) into an error, in which case you get the full traceback.

Reply:
If you add `error:invalid value encountered in less_equal:RuntimeWarning:scipy.stats._distn_infrastructure` to setup.cfg it will raise.  You can be as specific as you want with the error message part to only catch nans in certain ops.

Reply:
surpressing the warning looks the wrong solution to me.

If we have warnings in the test suite, then they should be fixed/surpressed there.
If user has nans in the data, then we don't want to surpress them.

just looking at the lines above, it seems that we are computing too many terms. 
Isn't there a truncation smaller than len(x) + 1?

Reply:
For this PR, reverting this accidental change is clearly the way to go.  In a dedicated PR we can look at which scipy.stats warnings we want to keep/suppress/re-issue-with-additional-info